core/services/fluxmonitorv2/contract_submitter.go

```diff
@@ -46,8 +48,8 @@ func NewFluxAggregatorContractSubmitter(
 
 // Submit submits the answer by writing a EthTx for the bulletprooftxmanager to
 // pick up
-func (c *FluxAggregatorContractSubmitter) Submit(roundID *big.Int, submission *big.Int) error {
-	fromAddress, err := c.keyStore.GetRoundRobinAddress()
+func (c *FluxAggregatorContractSubmitter) Submit(db *gorm.DB, roundID *big.Int, submission *big.Int) error {
+	fromAddress, err := c.keyStore.GetRoundRobinAddress() // TODO: should accept a db too
```

Comment:
It absolutely should, why not do it here?

Reply:
whoops meant to come back to this

Reply:
done